src/conversations/Conversations.ts

```diff
@@ -24,50 +29,35 @@ export default class Conversations {
    * List all conversations with the current wallet found in the network, deduped by peer address
    */
   async list(): Promise<Conversation[]> {
-    const messages = await this.client.listIntroductionMessages()
-    const seenPeers: Set<string> = new Set()
-    for (const message of messages) {
-      // Ignore all messages without sender or recipient address headers
-      // Makes getPeerAddress safe
-      if (!messageHasHeaders(message)) {
-        continue
-      }
+    const seenPeers = await this.getIntroductionPeers()
 
-      const peerAddress = this.getPeerAddress(message)
+    const conversations: Conversation[] = []
+    seenPeers.forEach((sent, peerAddress) =>
+      conversations.push(new ConversationV1(this.client, peerAddress, sent))
+    )
 
-      if (peerAddress) {
-        seenPeers.add(peerAddress)
-      }
+    for (const sealed of await this.client.listInvitations()) {
```

Comment:
We could use a `Promise.all()` to get the invitations and the intros in parallel. Would speed up this method a bit.

Reply:
Hm, not quite sure what you had in mind here. Do you mean something like this at the top?
```
  const [seenPeers, invitations] = await Promise.all([
      this.getIntroductionPeers(),
      this.client.listInvitations(),
    ])
```

Reply:
Just implemented it